### ISSUE TITLE
Cannot resolve google.cloud imports

### ISSUE THREAD

**Issue by velikodniy** [microsoft/pyright#4943]
**Describe the bug**

There's a regression in the latest release 1.1.302 where `google.cloud` imports are not resolved.

See also https://github.com/microsoft/pyright/issues/4255

**To Reproduce**

Install `google-cloud-storage` and Pyright:

```bash
python3.10 -m pip install google-cloud-storage==2.8.0 pyright==1.1.302
```

(I also tested with an npm-installed Pyright.)

Create a Python file `test.py` importing `google.cloud.storage`:

```python
from google.cloud import storage
```

Run Pyright:

```bash
pyright test.py
```

It will show the following error message:

```
No configuration file found.
No pyproject.toml file found.
Assuming Python platform Darwin
Searching for source files
Found 1 source file
pyright 1.1.302
/private/tmp/pyright/test.py
  /private/tmp/pyright/test.py:1:26 - error: "storage" is unknown import symbol (reportGeneralTypeIssues)
1 error, 0 warnings, 0 informations
Completed in 0.59sec
```

**Expected behavior**

No errors.

**Screenshots or Code**

<img width="850" alt="image" src="https://user-images.githubusercontent.com/1735272/231432905-e2eb5172-a740-45cc-aedd-97e743069c85.png">

**VS Code extension or command-line**

I use Pyright mostly as a CLI tool. Though, there's the same issue in VS Code (Python v2023.6.0, Pylance v2023.4.10).

**Additional context**

The issue is relevant for (at least):

- Pyright 1.1.301—1.1.303
- Pyright installed with either `pip` or `npm`.

**Comment by erictraut** [comment]
Thanks for the bug report, and apologies for the regression.

This was introduced as part of [this bug fix](https://github.com/microsoft/pyright/issues/4827). The fix didn't take into account namespace packages that include traditional packages, as is seen with `google.cloud.storage`.

This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.304, which is now published. It will also be included in a future release of pylance.

**Comment by guidopetri** [comment]
@erictraut this issue seems to be present for me again in 1.1.333 with `strict` enabled. Did we regress again?

**Comment by erictraut** [comment]
@guidopetri, I don't think there were any change between 1.1.332 and 1.1.333 that would affect import resolution. Furthermore, we have a regression test in place for this issue, so it's unlikely that the same issue would go undetected. If you think that there is a bug, please open a new issue with the repro steps (including library versions).
